### PR TITLE
Geofencing: Boot receiver to restore monitored geofences from disk

### PR DESCRIPTION
**Pull request by evan-masseau**
# Description

When a device reboots, the system location services will clear all geofences from memory. So this adds a boot receiver service to receive an intent when device reboots, check if app has geofencing level permissions, and re-add the geofences from storage.

## Due Diligence

- [ ] I have tested this on an emulator and/or a physical device.
- [ ] I have added sufficient unit/integration tests of my changes.
- [ ] I have adjusted or added new test cases to team test docs, if applicable.
- [ ] I am confident these changes are compatible with all Android versions the SDK currently supports.

## Release/Versioning Considerations

- [ ] `Patch` Contains internal changes or backwards-compatible bug fixes.
- [ ] `Minor` Contains changes to the public API. 
- [ ] `Major` Contains **breaking** changes.
- [ ] Contains readme or migration guide changes.
  - If so, please merge to a feature branch so documentation updates only go live upon official release.
- [ ] This is planned work for an upcoming release.
  - If no, author or reviewer should account for this in a release plan, or describe why not below.

## Related Issues/Tickets
 
 
 

---

> [!NOTE]
> Adds a boot receiver and LocationManager logic to re-register stored geofences after device reboot, with manifest updates and tests.
> 
> - **Location SDK (Geofencing)**:
>   - **Boot handling**:
>     - Add `KlaviyoBootReceiver` to handle `BOOT_COMPLETED` and delegate to `Registry.locationManager.restoreGeofencesOnBoot`.
>     - Implement `restoreGeofencesOnBoot(context)` in `KlaviyoLocationManager` to check permissions, ensure config init, load stored geofences, and re-register them.
>     - Extend `LocationManager` interface with `restoreGeofencesOnBoot(context)`.
>   - **Manifest**:
>     - Request `android.permission.RECEIVE_BOOT_COMPLETED` and register `com.klaviyo.location.KlaviyoBootReceiver` with `BOOT_COMPLETED` intent filter.
>   - **Tests**:
>     - Add `KlaviyoBootReceiverTest` to verify BOOT intent handling.
>     - Add/expand boot-restore tests in `KlaviyoLocationManagerTest` to cover permission checks, empty storage, and successful re-registration.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 10b9f482d43ab53e176c98901213bf0cbe912291. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationManager.kt [hunk 1]
```diff
@@ -396,4 +397,42 @@ internal class KlaviyoLocationManager(
...
+
+        Registry.log.info("Handling device boot event to restore geofences")
+
+        if (!Registry.isRegistered<Config>()) {
+            // Initialize Klaviyo with config to access dataStore
+            Klaviyo.registerForLifecycleCallbacks(context)
+        }
+
+        // Re-register geofences with the system
+        getStoredGeofences()
```
Comment (ajaysubra) [inline sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationManager.kt]: Should we pull from the API instead incase something has changed? I guess when app is launched this happens anyways?

Comment (evan-masseau) [reply sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationManager.kt]: Actually, I don't think I can do that. In this boot receiver, they either won't have provided a company ID yet, or else they'll have called `registerGeofencing` in their application launch code which will do the same thing anyway.

File: sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationBootReceiver.kt [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * BroadcastReceiver that handles device boot events to re-register geofences
+ * when the device reboots. System geofences are cleared on reboot, so we need
+ * to restore them from persistent storage.
+ */
+class KlaviyoLocationBootReceiver : BroadcastReceiver() {
+    override fun onReceive(context: Context, intent: Intent) {
+        if (intent.action == Intent.ACTION_BOOT_COMPLETED) {
+            try {
+                Registry.locationManager.restoreGeofencesOnBoot(context)
+            } catch (e: Throwable) {
```
Comment (dan-peluso) [inline sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationBootReceiver.kt]: Do we want `Throwable`

Comment (evan-masseau) [reply sdk/location/src/main/java/com/klaviyo/location/KlaviyoLocationBootReceiver.kt]:
```suggestion
            } catch (e: Exception) {
```
